### PR TITLE
More Whitesands Ruins

### PR DESCRIPTION
**Pull request by Maldaris**
## About The Pull Request

Fixes a bug with Whitesands Atmos causing explosive decompressions.
Updates to our latest version of extools
Ports 6 or so Lavaland ruins to Whitesands
Adds three variants of Survivor simple mobs, which wear the suit/guns created by 1GlitchyCent. 
Adds Survivor corpse spawners, a basic loot table, and loot spawners

Adds a "camp" system, which associates all mobs in the `map_template` definition with each-other using a randomized faction ID. Opens up further customizations in the future for this subset of ruins.

Adds the following camps:
* Survivor Camp
* Hunter Camp
* Gunslinger Camp
* Survivor Adobe Camp

## Why It's Good For The Game

Content!

## Changelog
:cl:
add: "Camp" system, which associates mobs within the camp with each other using a unique faction ID
add: Whitesands Camps & simple_mobs
fix: Fixes a bug with Whitesands Atmos causing explosive decompressions
/:cl:

### CODE REVIEW

File: code/controllers/subsystem/mapping.dm [hunk 1]
```diff
@@ -118,11 +121,14 @@ SUBSYSTEM_DEF(mapping)
...
 			spawn_rivers(ice_z, 4, /turf/open/lava/plasma/ice_moon, /area/icemoon/underground/unexplored)
-
+	// WS Edit Start - Whitesands
 	var/list/sand_ruins = levels_by_trait(ZTRAIT_SAND_RUINS)
 	if (sand_ruins.len)
 		seedRuins(sand_ruins, CONFIG_GET(number/whitesands_budget), list(/area/whitesands/surface/outdoors/unexplored), sand_ruins_templates)
-
+	var/list/sand_camps = levels_by_trait(ZTRAIT_SAND_CAMPS)
+	if (sand_camps.len)
+		seedRuins(sand_camps, CONFIG_GET(number/whitesands_budget), list(/area/whitesands/surface/outdoors/explored), sand_camps_templates)
```
Comment (MarkSuckerberg) [inline code/controllers/subsystem/mapping.dm]:
```suggestion
		seedRuins(sand_camps, CONFIG_GET(number/whitesands_budget), list(/area/whitesands/surface/outdoors/explored), sand_camps_templates)
```

File: code/__DEFINES/maps.dm [hunk 2]
```diff
@@ -84,6 +85,7 @@ require only minor tweaks.
     ZTRAIT_MINING = TRUE, \
 	ZTRAIT_TEMPERATURE_GRADIENT = TRUE, \
     ZTRAIT_SAND_RUINS = TRUE, \
+	ZTRAIT_SAND_CAMPS = TRUE, \
```
Comment (MarkSuckerberg) [inline code/__DEFINES/maps.dm]:
```suggestion
```

File: code/__DEFINES/maps.dm [hunk 3]
```diff
@@ -39,6 +39,7 @@ require only minor tweaks.
 #define ZTRAIT_ICE_RUINS "Ice Ruins"
 #define ZTRAIT_ICE_RUINS_UNDERGROUND "Ice Ruins Underground"
 #define ZTRAIT_SAND_RUINS "Sand Ruins" //Wasp edit - Whitesands
+#define ZTRAIT_SAND_CAMPS "Sand Camps" // WS Edit - Whitesands - Specifically used for non-ruin spawns, like mob camps/fauna nests
```
Comment (MarkSuckerberg) [inline code/__DEFINES/maps.dm]:
```suggestion
```

File: code/modules/mapping/ruins.dm [hunk 4]
```diff
@@ -123,8 +123,13 @@
...
 									break outer
-
-				placed_turf = current_pick.try_to_place(target_z,whitelist,forced_turf)
+				// WS Edit Start - Whitesands Camp Ruins
+				var/datum/map_template/ruin/camp/whitesands/C = current_pick
+				if (istype(C))
+					placed_turf = C.try_to_place(target_z, whitelist, forced_turf)
+				else
+					placed_turf = current_pick.try_to_place(target_z,whitelist,forced_turf)
+				// WS Edit End - Whitesands Camp Ruins
```
Comment (MarkSuckerberg) [inline code/modules/mapping/ruins.dm]:
```suggestion

				placed_turf = current_pick.try_to_place(target_z,whitelist,forced_turf)
```

**Review by MarkSuckerberg — CHANGES_REQUESTED** [review]
commit if you need em, don't if you've got plans for the code

File: code/datums/map_config.dm [hunk 5]
```diff
@@ -3,7 +3,7 @@
 //defaults to box
 //  -Cyberboss
 
-GLOBAL_LIST_INIT(mining_maps, list("lavaland" = 1, "icemoon" = 1, "whitesands" = 1, "random" = 0))
+GLOBAL_LIST_INIT(mining_maps, list("lavaland" = 0, "icemoon" = 0, "whitesands" = 1, "random" = 0))
```
Comment (MarkSuckerberg) [inline code/datums/map_config.dm]: ?

**Review by MarkSuckerberg — APPROVED** [review]

**Comment by MarkSuckerberg** [comment]
would still like clarification of change before merge

File: code/datums/map_config.dm (hunk 5 above)
Comment (Maldaris) [reply code/datums/map_config.dm]: whoops, debug code getting into main

Comment (Maldaris) [reply code/datums/map_config.dm]: fixed

**Pull request merged**